framework/src/org/apache/cordova/CordovaPlugin.java

```diff
@@ -132,6 +137,35 @@ public boolean execute(String action, JSONArray args, CallbackContext callbackCo
      * @return                Whether the action was valid.
      */
     public boolean execute(String action, CordovaArgs args, CallbackContext callbackContext) throws JSONException {
+        if (methodsMap == null) {
+            methodsMap = new HashMap<String, Method>();
+            for (Method method : this.getClass().getDeclaredMethods()) {
+                CordovaMethod cordovaMethod = method.getAnnotation(CordovaMethod.class);
+                if (cordovaMethod != null) {
+                    try {
```

Comment:
IMHO `try should be outside for for 2 reasons:
- if one the method fails then rest of them will as well.